src/ui/components/search/searchcomponent.js

```diff
@@ -344,59 +374,96 @@ export default class SearchComponent extends Component {
     }
   }
 
-  requestIconAnimationFrame (iconState) {
+  requestIconAnimationFrame (iconState, iconElClass) {
     if (this.iconState === iconState) {
       return;
     }
+    const prevState = this.iconState;
     this.iconState = iconState;
     if (!this.isRequestingAnimationFrame) {
       this.isRequestingAnimationFrame = true;
       window.requestAnimationFrame(() => {
-        this.forwardIcon.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
-        this.reverseIcon.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
+        const iconEls = DOM.queryAll(this._container, '.js-yxt-SearchBar-Icon');
+        iconEls.forEach((el) => {
+          el.classList.contains('js-yxt-SearchBar-LoadingIndicator')
+            ? el.classList.add('yxt-SearchBar-LoadingIndicator--inactive')
+            : el.classList.add('yxt-SearchBar-Icon--inactive');
+        });
+        const iconEl = DOM.query(this._container, iconElClass);
+        iconEl.classList.contains('js-yxt-SearchBar-LoadingIndicator')
+          ? iconEl.classList.remove('yxt-SearchBar-LoadingIndicator--inactive')
+          : iconEl.classList.remove('yxt-SearchBar-Icon--inactive');
+
+        iconEl.classList.remove('yxt-SearchBar-AnimatedIcon--paused');
         if (this.iconState === IconState.MAGNIFYING_GLASS) {
-          this.forwardIcon.classList.remove('yxt-SearchBar-AnimatedIcon--inactive');
-          this.reverseIcon.classList.add('yxt-SearchBar-AnimatedIcon--inactive');
-        } else if (this.iconState === IconState.YEXT) {
-          this.forwardIcon.classList.add('yxt-SearchBar-AnimatedIcon--inactive');
-          this.reverseIcon.classList.remove('yxt-SearchBar-AnimatedIcon--inactive');
+          const forwardEl = DOM.query(this._container, iconElClass).firstElementChild;
+          if (prevState === IconState.DONE_LOADING) {
+            forwardEl.classList.remove('Icon--yext_animated_forward');
+            forwardEl.classList.add('Icon--magnifying_glass');
+          } else {
+            forwardEl.classList.add('Icon--yext_animated_forward');
+            forwardEl.classList.remove('Icon--magnifying_glass');
+          }
         }
         this.isRequestingAnimationFrame = false;
       });
     }
   }
 
+  animateIconToLoading () {
+    this.loadingIconUrl
+      ? this.requestIconAnimationFrame(IconState.LOADING, '.js-yxt-SearchBar-loadingImage')
+      : this.requestIconAnimationFrame(IconState.LOADING, '.js-yxt-SearchBar-LoadingIndicator');
```

Comment:
One of these has loading with a lowercase 'l' and the other has loading with an uppercase 'L'. Is it possible to be consistent with that?

Reply:
ah I saw in the existing code that the custom images have `buttonImage` lowercase while the animated ones have uppercase `AnimatedIcon` so I was kinda following it. I could change them all to uppercase if that's preferred?

Reply:
I think i'd prefer to have them consistent within this file. How about CustomLoadingImage?